test/devfiles/v200/commands/ExecMissingEnvOrder.yaml

```diff
@@ -0,0 +1,16 @@
+  - id: testExec1
+    exec:
+      attributes: 
+        test: Exec Test test
+        scope: Api
+      commandLine: 'echo "Hello ${GREETING} ${USER}"'
+      component: api-tests
+      group: 
+        isDefault: true
+        kind: test
+      hotReloadCapable: false
+      label: "Command Exec test"
+      env:
+        - value : "USER"
+          name : USER
```

Comment:
I'm not sure what this test is checking.

Reply:
May be a redendant one, just checking it does not matter which order you put the name and value in.

Reply:
I fixed the issues you raised and also one I noticed. The easiest way to review is to go to the branch and then the test subdirectory which the PR creates: https://github.com/mmulholla/api/tree/AddAPITests/test

There you can see the README and easily browse the new files.

The tests may seem a bit pointless right now, but will be useful in the future when the schema is updated and we check if v200 devfiles still work with the new schema.

Reply:
Also not sure how these tests are over complicated. They perform two goals :
 
1. test every attriibute in the schema, make sure mandatory attributes are indeed mandatory and optional ones are indeed optional. Then where attributes have a restriction make sure those restrictions are enforced. 
2. Make it easy to add new tests for a new attribute or different schema.

Add to this that the validator only reports the first problem, so you need a test for every error case. There is obviously a bunch of different ways you can achieve these goals but, would be equally complicated.

Adding a fuzzer to generate values would be interesting but would need all the logic in the go code which then makes it more difficult to add and changes tests. Also not sure how it works when an attribute is restricted to a range of values.

Maybe i misunderstand how the fuzzer would be used.